qiskit/visualization/pulse/matplotlib.py

```diff
@@ -59,18 +59,20 @@ def __init__(self, t0, tf):
         self._labels = None
         self.enable = False
 
-    def add_instruction(self, start_time, pulse):
+    def add_instruction(self, start_time, instruction):
         """Add new pulse instruction to channel.
 
         Args:
             start_time (int): Starting time of instruction
-            pulse (Instruction): Instruction object to be added
+            instruction (Instruction): Instruction object to be added
         """
-
+        if isinstance(instruction, Delay):
+            return
+        pulse = getattr(instruction, 'pulse', instruction.command)
```

Comment:
`ShiftPhase` and `SetFrequency` don't have `.command` and this seems to return `None`. This is how previous bug was caused.

Reply:
Ah, I'm really unfamiliar with this, thanks. from the name of the function and the argument names it seemed like this wouldn't process non pulse instructions. I'll update

Reply:
I'm wrong -- the name of the function should have made it clear